linker/linker.cpp

```diff
@@ -882,7 +890,10 @@ static const ElfW(Sym)* dlsym_linear_lookup(android_namespace_t* ns,
 soinfo* find_containing_library(const void* p) {
   ElfW(Addr) address = reinterpret_cast<ElfW(Addr)>(p);
   for (soinfo* si = solist_get_head(); si != nullptr; si = si->next) {
-    if (address >= si->base && address - si->base < si->size) {
+    if ((address >= si->base && address - si->base < si->size)
+        || si->find_rand_segment_info(address) != nullptr) {
+      // FIXME: this assumes that si->size also covers
```

Comment:
It might have been that at first I added the comment, then later went back in and added the check. I need to check the commit history for that.